test/libsolidity/semanticTests/constants/regression_8406.sol

```diff
@@ -0,0 +1,12 @@
+// Test for regression of https://github.com/ethereum/solidity/issues/8406
```

Comment:
Can you change the name of the test to be more descriptive?

Reply:
Renamed to `asm_address_constant_regression`.